### PR TITLE
xar: 1.6.1 -> 498

### PR DESCRIPTION
**Pull request by tie**
## Description of changes

This change switches the xar package from unmaintained fork of the original project to the Apple Open Source tarball. We also add some patches from Debian/Gentoo on top, but otherwise the patch set is original and fixes some longstanding bugs (i.e. I’ve spent a considerable amount of time to ensure that all tests provided in the tarball are passing). If merged, Nixpkgs would probably be the only distribution that actually runs `xar` tests…

See also https://repology.org/project/xar/versions

## Things done

- Built on → for platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [x] aarch64-linux → aarch64-linux (pkgsMusl)
  - [x] aarch64-linux → aarch64-linux (pkgsStatic)
  - [x] aarch64-linux → x86_64-linux (pkgsCross.gnu64)
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
 Built `darwin.apple_sdk.MacOSX-SDK`.
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by tie** [comment]
Mostly ready for review, but there are some test failures because of mismatching mtime in `pkgsMusl.xar.tests.integrationTest` (that is, with musl in particular).

File: pkgs/by-name/xa/xar/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/xa/xar/package.nix:90]:
```suggestion
    ++ lib.optional withExt2 e2fsprogs
    ++ lib.optional withAcl acl
    ++ lib.optional withBzip2 bzip2
    ++ lib.optional withLzma xz
    ++ lib.optional stdenv.hostPlatform.isMusl musl-fts;
```
nit

File: pkgs/by-name/xa/xar/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/xa/xar/package.nix:79]: If there would be more linux/darwin specific packages it would probably be better to add them like the musl one

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix]: Done.

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix:79]: Hm, are you suggesting to use `lib.optional` in this case? 🤔

I’m not really sure about this, `musl-fts` is an additional platform-specific dependency, while OpenSSL and CommonCrypto is a platform-specific choice between different cryptography libraries that I feel like is better expressed with `if … then … else` compare to two distinct `lib.optional`s. That said, I don’t have a strong preference for either.

Comment (emilazy) [reply pkgs/by-name/xa/xar/package.nix:79]: In a matter of months I believe `CommonCrypto` will be available without any dependency on macOS as part of the SDK rework, so any colour of bikeshed is fine here.

Comment (tie) [reply pkgs/by-name/xa/xar/package.nix:79]: Would be nice to use `CommonCrypto` for non-Darwin platforms as well since it looks like OpenSSL support was abandoned by Apple. Gentoo does the opposite and always uses OpenSSL even on Darwin.
https://gitweb.gentoo.org/repo/gentoo.git/tree/app-arch/xar/xar-1.8.0.0.498.ebuild?id=f09efc0fdb5cfa3271f628d0d8508e786fecab50#n57

**Comment by tie** [comment]
W.r.t. musl test failures: we need a recent version of musl for `%F` time format directive (https://git.musl-libc.org/cgit/musl/commit/src/time/strptime.c?id=fced99e93daeefb0192fd16304f978d4401d1d77). See also #229439.
I’ll add a patch to use explicit `%Y-%m-%dT%H:%M:%SZ` format instead.

**Comment by tie** [comment]
Fixed pkgsMusl for xar (and also pbzx), verified with
```
nix build --file . pkgsMusl.darwin.apple_sdk.MacOSX-SDK pkgsMusl.xar.tests
```

**Comment by tie** [comment]
@SuperSandro2000, updated to use `lib.optional` for OpenSSL/CommonCrypto as you’ve suggested in https://github.com/NixOS/nixpkgs/pull/329721#discussion_r1694321150.
Also added myself to the maintainers list.

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [inline pkgs/by-name/xa/xar/package.nix:66]: Oops, forgot to change these lines after debugging.

Comment (tie) [reply pkgs/by-name/xa/xar/package.nix]: Fixed.

**Comment by tie** [comment]
@SuperSandro2000 @emilazy gentle ping, the PR is ready for another round of reviews.

File: pkgs/by-name/xa/xar/package.nix
Comment (emilazy) [inline pkgs/by-name/xa/xar/package.nix:52]: I guess this can’t use `fetchFromGitHub` for bootstrap reasons? It might be nice to use GitHub’s tarball URL, though, for more of a guarantee that it matches the Git repository. GitHub seems to be Apple’s main FOSS distribution method at this point anyway.

File: pkgs/by-name/xa/xar/package.nix
Comment (emilazy) [inline pkgs/by-name/xa/xar/package.nix:165]: How is this going to work when we’re `fetchurl`ing a tarball?

**Review by emilazy — COMMENTED** [review]
I haven’t reviewed the patches because there’s so much. It makes me a bit uneasy to have this many, but it sounds like you were thorough about it. Have you considered forking the upstream repository so that other distros can source these fixes more easily and we don’t have to maintain them as a set of raw `.patch` files? With this many changes you’re effectively our new upstream for the package anyway :)

File: pkgs/by-name/xa/xar/package.nix
Comment (reckenrode) [reply pkgs/by-name/xa/xar/package.nix:52]: `fetchFromGitHub` is allowed in the Darwin bootstrap because it’s really a `fetchzip`. It’s obly `fetchgit` that’s a problem because it depends on git, which is not in the bootstrap tools.

File: pkgs/by-name/xa/xar/package.nix
Comment (reckenrode) [reply pkgs/by-name/xa/xar/package.nix]: If OpenSSL works on Darwin, I see no reason not to use it on Darwin. The Darwin stdenv bootstrap already requires OpenSSL due to ld64 (if not other packages, but I know ld64 does for sure because I wrote it that way).

Comment (emilazy) [reply pkgs/by-name/xa/xar/package.nix]: Seems reasonable to me if we’re having to maintain OpenSSL support for Linux anyway. Is using `CommonCrypto` on Linux at all viable, though, if that lets us defer maintenance to Apple?

File: pkgs/by-name/xa/xar/package.nix
Comment (reckenrode) [inline pkgs/by-name/xa/xar/package.nix:171]: In the Darwin refactor, I’ve been changing source release maintainers to `lib.teams.darwin.members` because the existing maintainers on them are inactive. Using the Darwin team prevents that (assuming the members of that team are not themselves inactive, but that’s easily fixable).

File: pkgs/by-name/xa/xar/package.nix
Comment (reckenrode) [reply pkgs/by-name/xa/xar/package.nix:21]: `availableOn stdenv.hostPlatform` should be true on Darwin then, shouldn’t it?

Comment (emilazy) [reply pkgs/by-name/xa/xar/package.nix:21]: Yes. My point is that both packages are `lib.platforms.unix`, so you might as well write `true` instead (and then perhaps not write the flag at all).

**Comment by reckenrode** [comment]
> Have you considered forking the upstream repository so that other distros can source these fixes more easily and we don’t have to maintain them as a set of raw .patch files?

There are two downsides to maintaining a fork:

1. It’s effectively its own project. If the repo is public, people will submit issues and PRs. That’s potentially more work. I opted not to do that with my cctools and ld64 for that reason.
2. Apple releases code dumps, which become increasingly difficult to integrate as patches diverge your fork from upstream (see also cctools-port, which is having that problem). It’s easier to rebase patches on top of a new dump, but you can’t do that in a repo due to rewriting history.

File: pkgs/by-name/xa/xar/package.nix
Comment (reckenrode) [reply pkgs/by-name/xa/xar/package.nix]: CommonCrypto depends on corecrypto, which is extremely non-free.

Comment (reckenrode) [reply pkgs/by-name/xa/xar/package.nix]: As far as I can tell, xar is using CommonCrypto for hashing. It’s the same situation as ld64, but the work has already been done to support using OpenSSL for hashing (rather than having to write compatibility shims like I did).

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix:73]: Yes, but I believe that in this case it’s actually an issue with Nixpkgs package because it’s namespaced twice, i.e. `include/libxml2/libxml/xmlwriter.h` instead of `include/libxml/xmlwriter.h`…

The libxml’s project documentation uses `#include <libxml/xmlwriter.h>` in examples as well and that just doesn’t work with C compiler wrapper and hooks in Nixpkgs because of that unnecessary namespacing (because stdenv only adds top-level `include` directory for each package to the C header search path).

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix:165]: `meta.homepage` takes precedence over `src`.

Interface:
https://github.com/NixOS/nixpkgs/blob/ab16e260852059740a6ee5d87a506cf918735bc1/pkgs/common-updater/git-updater.nix#L14-L16

Implementation:
https://github.com/NixOS/nixpkgs/blob/164d4463dfeda1c15594e72e60a2c7e80962aa7c/pkgs/common-updater/scripts/list-git-tags#L41-L46

Not sure if gitUpdater is documented in Nixpkgs manual though, ^F didn’t give any results (but perhaps I was looking in the wrong place).

That said, thanks for pointing that out, I’ve added `updateScript` and completely forgot to test that it actually works. Probably needs `rev-prefix` because tags are prefixed with `xar-`.

**Comment by tie** [comment]
>Have you considered forking the upstream repository so that other distros can source these fixes more easily and we don’t have to maintain them as a set of raw .patch files?

Using patches instead of a separate repository helped me to focus on fixing bugs while keeping the changes minimal. Ideally, some of these patches should be submitted to Apple, but I couldn’t find the intended communication channel (if there is any). Perhaps an email to `opensource@apple.com`?

>There are two downsides to maintaining a fork:

In addition to that, `xar` is _technically_ deprecated, and the number of patches needed to build and run it with modern hardware and software unfortunately reflects that.

```console
$ xar --help
DEPRECATION WARNING:
	xar is no longer under active development by Apple.
	Clients of xar should pursue alternative archive formats.

Usage: …
```

Other distributions apply a similar set of patches, but, as far as I’m aware, none test `xar` on all/any platforms and I happened to use a fairly common `aarch64-linux` platform where `xar` was broken :shrug:

Also Debian seems to be OK with using patches from Gentoo (see https://salsa.debian.org/debian/xar/-/tree/dda6fee008961942a2578e24aaa3b1ef7de795d1/debian/patches), so I don’t think it’d be an issue for other distributions to use patches from Nixpkgs.

>It’s easier to rebase patches on top of a new dump, but you can’t do that in a repo due to rewriting history.

Exactly, and because of history rewrites there is not much of a difference between a patch set and a Git repository that uses branch per upstream tag (like [cctools-port](https://github.com/tpoechtrager/cctools-port) does).

That said, if decide to fork the repository, it should be done under the [NixOS](https://github.com/NixOS) organization. After all, Apple itself forked the original `xar` project because it was a hassle to maintain patches.
Excerpt from [Apple’s xar](https://github.com/apple-oss-distributions/xar) README:
>Patched folder, included in the project contain some patches on the original project xar-1.4.tar.gz. This project was created because creating changes ontop of the original open source project has become a serious hassle.

The patches in this PR can be applied on top of the Apple’s Git repository with `git am`.

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix:128]: To be fair, this is how `testers.testVersion` behaves as well since it uses `nativeBuildInputs` and idiomatic usage is to pass the package fixed point from `callPackage` arguments (so it’s spliced).

This seems like a reasonable compromise between using an emulator for tests and not running the tests at all (or failing to do so because we can’t execute the program built for a different platform). Alternatively, we can define tests under `optionalAttrs (buildPlatform.canExecute hostPlatform)`, but I haven’t seen this done in any package.

We can’t run _all_ `xar` tests in Nix sandbox anyway (e.g. on Linux, ACL/EAs syscalls are filtered with seccomp), hence we expose `tests.manualTest` with `xar` built for the host platform.

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix:21]: `acl` package is not available on macOS, but `e2fsprogs` indeed is.

> Is this for bootstrap purposes?

I wasn’t really sure what set of packages was available for darwin stdenv bootstrap so I added a feature flag for each optional dependency. Turns out it builds fine, so unless someone is working on reducing the bootstrap chain for Darwin, we can drop these flags.

>unless you have ambitions of getting this to run on Windows

I actually tried building for Windows, but the code at least needs `uid_t` and that’s a sign of a potential rabbit hole so I didn’t investigate further.

Comment (emilazy) [reply pkgs/by-name/xa/xar/package.nix:21]: Yeah, I started looking into getting `acl` to build but it wasn’t clear if we need it (does Darwin just have those headers natively?).

If there’s no bootstrap issues I’d personally prefer dropping these flags as most likely being gratuitous API surface. Reducing the bootstrap chain would be noble, but is probably a different project altogether.

File: pkgs/by-name/xa/xar/package.nix
Comment (emilazy) [reply pkgs/by-name/xa/xar/package.nix:73]: Uh, yeah, that does indeed sound messed up and we should fix it. Sorry for assuming something more normal was going on here.

I guess this happens to work because most things use `pkg-config` to find the headers?

File: pkgs/by-name/xa/xar/package.nix
Comment (emilazy) [reply pkgs/by-name/xa/xar/package.nix:128]: Fair enough. I’d personally lean towards avoiding a false sense of security from testing a version that isn’t actually to be deployed – especially since it results in an additional build of `xar` (albeit one that is likely cached) – but if it’s common for other packages to be doing this kind of thing and we don’t have an ideal solution I certainly won’t block on it.

It would be nice if we could just make the tests native to the host platform but condition them on the availability of a builder for that platform, which is more nuanced than `buildPlatform.canExecute`, but I guess that would be pretty impure.

File: pkgs/by-name/xa/xar/package.nix
Comment (emilazy) [reply pkgs/by-name/xa/xar/package.nix:52]: Right. Then I think it’d be nice to use `fetchFromGitHub` here.

File: pkgs/by-name/xa/xar/package.nix
Comment (emilazy) [reply pkgs/by-name/xa/xar/package.nix:165]: It uses `src.meta.homepage`, which is different to `meta.homepage` – that’s how `fetchFromGitHub` etc. expose their repository URLs. So I don’t think this will actually work, unless I’m misunderstanding something.

What do you think about using `fetchFromGitHub` and `nix-update-script` instead? That would use the GitHub API directly for updates and also ensure that the sources match what’s in the repository.

File: pkgs/by-name/xa/xar/package.nix
Comment (emilazy) [reply pkgs/by-name/xa/xar/package.nix]: Perhaps let’s use OpenSSL on every platform, then, if we’re having to maintain that support anyway and `CommonCrypto` isn’t a option. Not really a big deal either way though.

**Comment by emilazy** [comment]
> Using patches instead of a separate repository helped me to focus on fixing bugs while keeping the changes minimal. Ideally, some of these patches should be submitted to Apple, but I couldn’t find the intended communication channel (if there is any). Perhaps an email to `opensource@apple.com`?

Good luck with that 🫠 I don’t think Apple is really interested in outside contributions, unfortunately.

> Also Debian seems to be OK with using patches from Gentoo (see https://salsa.debian.org/debian/xar/-/tree/dda6fee008961942a2578e24aaa3b1ef7de795d1/debian/patches), so I don’t think it’d be an issue for other distributions to use patches from Nixpkgs.

Yes, I don’t think it’d be an issue per se, it’d just be unfortunate for other distributions to have to copy around so many patches and keep them in sync (more likely they’d bitrot, I reckon).

> Exactly, and because of history rewrites there is not much of a difference between a patch set and a Git repository that uses branch per upstream tag (like [cctools-port](https://github.com/tpoechtrager/cctools-port) does).

IMO the difference is that Git is better tooling to work with a bunch of stacked patches on top of an existing Git repository. The next person to work on updating these patches will probably start by recreating the exact Git repository that would otherwise be published with `git am`, like you said. If you’re more comfortable with the patch files, though, I’m okay with them. You are basically maintaining the upstream project now, though, for better or worse :)

> That said, if we decide to fork the repository, it should be done under the [NixOS](https://github.com/NixOS) organization. After all, Apple itself forked the original `xar` project because it was a hassle to maintain patches. Excerpt from [Apple’s xar](https://github.com/apple-oss-distributions/xar) README:

Most of the software we package comes from non‐NixOS repositories, but yeah, the downside is that we’d have to switch to another fork if the person taking care of the patches changes. GitHub does at least make that easy, though.

**Comment by emilazy** [comment]
Thanks for this impressive effort, by the way! Really nice to see clean‐ups like this.

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix:73]: >I guess this happens to work because most things use `pkg-config` to find the headers?

Yes, though some other packages set `NIX_CFLAGS_COMPILE` as well. E.g. https://github.com/NixOS/nixpkgs/blob/2e11a22606820ece544b1b7f14de9ae32392ef9a/pkgs/servers/sql/postgresql/generic.nix#L90-L92

Uh, yes, libxml2 also has its own `xml2-config` program that is similar to pkgconf/pkg-config and usually does not work when cross-compiling…

I’m not really sure if there is some knob in libxml2 build system to put the headers in `$dev/include` instead of `$dev/include/libxml2`, but at the very least it should be possible to just move the headers post install and add a symlink for compatibility.
```
mv $dev/include/libxml2 libxml2
rmdir $dev/include
mv libxml2 $dev/include
ln -s . $dev/include/libxml2
```

**Comment by reckenrode** [comment]
Also, sorry about causing the merge conflict. I did a search for PRs against the Darwin stdenv but missed this one. I reformatted the stdenv in https://github.com/NixOS/nixpkgs/pull/333962 because I anticipate doing quite a bit of cleanup while working on the SDK refactor, and it’s easier/nicer if I can work on a formatted stdenv.

**Comment by tie** [comment]
>Also, sorry about causing the merge conflict.

It’s OK, I’ll have to update the PR with suggested changes anyway and it’s just a single line change in stdenv for separate outputs.

File: pkgs/by-name/xa/xar/package.nix
Comment (reckenrode) [reply pkgs/by-name/xa/xar/package.nix:21]: I also tried to get `acl` to build a while back. The problem is the APIs for xattrs are different between Darwin and Linux. `attr` fails to build because it doesn’t support the Darwin ones.

For example, `listxattr` takes an options argument on Darwin but not Linux:
* https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/listxattr.2.html
* https://www.man7.org/linux/man-pages/man2/listxattr.2.html

Comment (tie) [reply pkgs/by-name/xa/xar/package.nix:21]: > does Darwin just have those headers natively?
I think it has native ACL implementation in Libc.
https://github.com/apple-oss-distributions/Libc/blob/Libc-1592.100.35/include/sys/acl.h

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix:165]: NB using `nix-update-script` is blocked by https://github.com/Mic92/nix-update/pull/269 that is not released yet.

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix]: Updated to use `fetchFromGitHub`.

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix]: Added `lib.teams.darwin.members` to the list of maintainers.

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix]: Done.

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix]: Since this conversation, I’ve seen a handful of packages that define tests under `optionalAttrs` so I went with your suggestion and wrapped the tests under `canExecute` for `finalAttrs.finalPackage`.

File: pkgs/by-name/xa/xar/package.nix
Comment (tie) [reply pkgs/by-name/xa/xar/package.nix]: The update script should work once Nixpkgs contains the next nix-update release. Tested after manually bumping `nix-update` to unreleased commit and rolling back `xar` to a previous release.

<details>

```diff
diff --git a/pkgs/by-name/ni/nix-update/package.nix b/pkgs/by-name/ni/nix-update/package.nix
index db47d4b5abd6..21f53318d0da 100644
--- a/pkgs/by-name/ni/nix-update/package.nix
+++ b/pkgs/by-name/ni/nix-update/package.nix
@@ -9,14 +9,14 @@
 
 python3.pkgs.buildPythonApplication rec {
   pname = "nix-update";
-  version = "1.5.0";
+  version = "unstable-2024-08-21";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "Mic92";
     repo = pname;
-    rev = version;
-    hash = "sha256-zWkJs6anDddTiUestnqGVfG1GErLJr5EarWI3DcKZpU=";
+    rev = "737121eccb67542e8c004c64da833fede2e80c64";
+    hash = "sha256-xn0dC4M3mfItxP+s3/v3Hz/CSKp74VH/gMfufKxl9/4=";
   };
 
   nativeBuildInputs = [
diff --git a/pkgs/by-name/xa/xar/package.nix b/pkgs/by-name/xa/xar/package.nix
index 4ad071f36a1f..c77b75b3c09c 100644
--- a/pkgs/by-name/xa/xar/package.nix
+++ b/pkgs/by-name/xa/xar/package.nix
@@ -31,13 +31,13 @@
 }:
 stdenv.mkDerivation (finalAttrs: {
   pname = "xar";
-  version = "498";
+  version = "496";
 
   src = fetchFromGitHub {
     owner = "apple-oss-distributions";
     repo = "xar";
     rev = "xar-${finalAttrs.version}";
-    hash = "sha256-RyWeR/ZnDBHIZhwzVxETdrTTPQA2VgsLZegRkxX1240=";
+    hash = "sha256-MpV93JstGuoKJJWhG3Tckv1q2N3mTVg60LSDAGqoCOQ=";
   };
 
   # Update patch set with
```
</details>

Running `nix develop --file maintainers/scripts/update.nix --argstr package xar` bumps the `xar` version and updates `hash` as expected.

Comment (tie) [reply pkgs/by-name/xa/xar/package.nix]: Marking this thread as resolved based on the comment above since nix-update release is non-blocking for the package implementation.

**Comment by tie** [comment]
Rebased, addressed previous review comments, ready for another round of reviews.

**Review by emilazy — APPROVED** [review]
Diff looks good, reproduced the build and tests (pure and impure) on `aarch64-linux` and `aarch64-linux`. Happy to merge once @reckenrode has taken another look and once someone reproduces the build on x86-64.

**Comment by reckenrode** [comment]
I haven’t run the tests or tried to build it, but the changes look good to me.

**Review by JohnRTitor — APPROVED** [review]
Built on x64-linux.

**Comment by reckenrode** [comment]
It’s defined at https://github.com/apple-oss-distributions/xar/blob/eaeb03d085970f94938dc6785cd86d22eaa54b47/xar/include/xar.h.in#L268. The availability annotation could be removed since nixpkgs will be shipping the same dylib regardless of deployment target.

**Comment by emilazy** [comment]
Seems reasonable. It shouldn’t cause anything worse than warnings to leave it in, though, right?

**Comment by reckenrode** [comment]
Probably shouldn’t cause any issues other than warnings unless one uses `-Werror`.

**Comment by emilazy** [comment]
Let’s merge this as‐is for now and we can make a decision about whether to patch the availability stuff out of the headers in another PR. Thank you for the great work as always, @tie!

**Pull request merged**
